### PR TITLE
8298893: Rename option UsePolyIntrinsics to UsePoly1305Intrinsics 

### PR DESCRIPTION
**Pull request by vpaprotsk**
Fix for better documentation. FYI @jnimeh

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8298893](https://bugs.openjdk.org/browse/JDK-8298893): Rename option UsePolyIntrinsics to UsePoly1305Intrinsics

### Reviewers
 * [Jamil Nimeh](https://openjdk.org/census#jnimeh) (@jnimeh - **Reviewer**) ⚠️ Review applies to [a4faddeb](https://git.openjdk.org/jdk20/pull/49/files/a4faddeb2d36eb82468a36e9f7def7c635e9c252)
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [a4faddeb](https://git.openjdk.org/jdk20/pull/49/files/a4faddeb2d36eb82468a36e9f7def7c635e9c252)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk20 pull/49/head:pull/49` \
`$ git checkout pull/49`

Update a local copy of the PR: \
`$ git checkout pull/49` \
`$ git pull https://git.openjdk.org/jdk20 pull/49/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 49`

View PR using the GUI difftool: \
`$ git pr show -t 49`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk20/pull/49.diff">https://git.openjdk.org/jdk20/pull/49.diff</a>

</details>

### CODE REVIEW

**Review by jnimeh — APPROVED** [review]
Thanks for making this change.  The changes look good to me.

**Comment by jnimeh** [comment]
When this is integrated I will take care of man page/doc changes for both our intrinsics.  Two other items of note:

- Please remember to update the release note with the modified name (JDK-8297970)
- You might need to modify this issue to be a P3 bug rather than P5 RFE in order for it to get backported into JDK 20, which we want it to be.

**Comment by vpaprotsk** [comment]
@jnimeh  
- Ah, I just noticed that @chhagedorn changed the issue overnight. Can change it back to P3 and against jdk20?
- yep, will change the release note, thanks for the reminder

**Comment by jnimeh** [comment]
I would think you'd need to make a backport for JDK 20.  Perhaps there you can set the backport to Bug/P3.

**Review by vnkozlov — COMMENTED** [review]
We are in RDP1 phase. Enhancement changes require approval by following process:
https://openjdk.org/jeps/3#Late-Enhancement-Request-Process

**Comment by vpaprotsk** [comment]
I think there is some confusion.. I had originally opened a "bug" against jdk20 (per a discussion with Jamil). Last night the issue got switched to a P5 enhancement against dev.

So as to not add to the confusion, not going to touch the issue or the PR until someone tells me otherwise? @jnimeh @vnkozlov @chhagedorn  ?

**Comment by vnkozlov** [comment]
@vpaprotsk 
Such renaming is enhancement. But these changes are simple and will be approved for JDK 20.
It is easy to follow process I pointed: add label and comment to [JDK-8298893](https://bugs.openjdk.org/browse/JDK-8298893). And wait approval.
Note, for documentation changes you don't need approval during RDP1.

**Comment by vpaprotsk** [comment]
Added a comment and labels to the JBS issue. I think thats it, but let me know what else I've missed or what else need fixin'

**Comment by vnkozlov** [comment]
> Added a comment and labels to the JBS issue. I think thats it, but let me know what else I've missed or what else need fixin'

Good. We have to wait when Mark approve it or delegate.

File: src/hotspot/share/runtime/globals.hpp [hunk 1]
```diff
@@ -238,7 +238,7 @@ const int ObjectAlignmentInBytes = 8;
   product(bool, UseBASE64Intrinsics, false,                                 \
           "Use intrinsics for java.util.Base64")                            \
                                                                             \
-  product(bool, UsePolyIntrinsics, false, DIAGNOSTIC,                       \
+  product(bool, UsePoly1305Intrinsics, false, DIAGNOSTIC,                       \
```
Comment (chhagedorn) [inline src/hotspot/share/runtime/globals.hpp]: You should fix the alignment of the `\` here.

**Review by chhagedorn — COMMENTED** [review]
> @vpaprotsk Such renaming is enhancement. But these changes are simple and will be approved for JDK 20. It is easy to follow process I pointed: add label and comment to [JDK-8298893](https://bugs.openjdk.org/browse/JDK-8298893). And wait approval. Note, for documentation changes you don't need approval during RDP1.

Sorry for the confusing and thanks @vnkozlov for jumping in and providing pointers to still get this into JDK 20. As Vladimir has mentioned, a flag renaming is an enhancement. But since it's low-risk, this should be easily approved for JDK 20.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me. Just FTR, I think such flag renamings are non-trivial and therefore require two reviews before integration.

File: src/hotspot/share/runtime/globals.hpp (hunk 1 above)
Comment (vpaprotsk) [reply src/hotspot/share/runtime/globals.hpp]: done. thanks for spotting this

**Review by chhagedorn — APPROVED** [review]

**Comment by vnkozlov** [comment]
This late enhancement request for JDK 20 was approved.

**Pull request closed without merging**